src/sage/combinat/symmetric_group_algebra.py

```diff
@@ -1941,6 +1941,46 @@ def _dft_seminormal(self, mult='l2r'):
         snb = self.seminormal_basis(mult=mult)
         return matrix([vector(b) for b in snb]).inverse().transpose()
 
+    def _dft_modular(self):
+        """
+        Return the discrete Fourier transform when the characteristic divides the order of the group.
+        
+        See [Mur1983]_ for the construction of central primitive orthogonal idempotents.
+        For each idempotent `e_i` we have a projection `v \\mapsto v*e_i`. This is a homomorphism.
+        We choose a basis for each submodule spanned by `\\{\\sigma*e_i | \\sigma \\in S_n\\}`.
+        The change-of-basis from the standard basis `\\{\\sigma\\}_\\sigma` is returned.
+
+        EXAMPLES::
+
+            sage: GF2S3 = SymmetricGroupAlgebra(GF(2),3)
+            sage: GF2S3.dft()
+            [1 0 0 0 1 0]
+            [0 1 0 0 0 1]
+            [0 0 1 0 0 1]
+            [0 0 0 1 1 0]
+            [1 0 0 1 1 0]
+            [0 1 1 0 0 1]
+        """
```

Comment:
Regarding the docstring discussion of `r"""` vs double backslashes: In the future raw strings will be the preferred approach (see #37508; just learned this myself), so it's probably a good idea to change it for the new code. I've adapted it as a suggestion below, so that you can simply commit it.
```suggestion
        r"""
        Return the discrete Fourier transform when the characteristic divides the order of the group.
        
        See [Mur1983]_ for the construction of central primitive orthogonal idempotents.
        For each idempotent `e_i` we have a projection `v \mapsto v*e_i`. This is a homomorphism.
        We choose a basis for each submodule spanned by `\{\sigma*e_i | \sigma \in S_n\}`.
        The change-of-basis from the standard basis `\{\sigma\}_\sigma` is returned.

        EXAMPLES::

            sage: GF2S3 = SymmetricGroupAlgebra(GF(2),3)
            sage: GF2S3.dft()
            [1 0 0 0 1 0]
            [0 1 0 0 0 1]
            [0 0 1 0 0 1]
            [0 0 0 1 1 0]
            [1 0 0 1 1 0]
            [0 1 1 0 0 1]
        """
```